units/task_stranded_cleanup.go

```diff
@@ -69,39 +74,54 @@ func (j *taskStrandedCleanupJob) Run(ctx context.Context) {
 		taskIDs = append(taskIDs, h.RunningTask)
 		hostIDs = append(hostIDs, h.Id)
 
-		j.AddError(model.ClearAndResetStrandedHostTask(&h))
+		catcher.Wrapf(model.ClearAndResetStrandedHostTask(&h), "fixing stranded host task '%s' on host '%s'", h.RunningTask, h.Id)
 	}
 
+	grip.InfoWhen(len(taskIDs) > 0 || len(hostIDs) > 0, message.Fields{
+		"message":   "fixed tasks stranded on already-terminated hosts",
+		"job":       j.ID(),
+		"op":        j.Type().Name,
+		"num_hosts": len(hosts),
+		"num_tasks": len(taskIDs),
+		"tasks":     taskIDs,
+		"hosts":     hostIDs,
+	})
+
+	return catcher.Resolve()
+}
+
+func (j *taskStrandedCleanupJob) fixTasksStuckDispatching() error {
 	tasks, err := task.FindStuckDispatching()
 	if err != nil {
-		j.AddError(err)
+		return errors.Wrap(err, "finding tasks that are stuck dispatching")
 	}
 
-	tasksToDeactivate := []task.Task{}
+	catcher := grip.NewBasicCatcher()
+	var tasksToDeactivate []task.Task
 	for _, t := range tasks {
-		if time.Since(t.CreateTime) >= 2*7*24*time.Hour {
+		if time.Since(t.ActivatedTime) >= task.UnschedulableThreshold {
 			tasksToDeactivate = append(tasksToDeactivate, t)
 		} else {
 			details := &apimodels.TaskEndDetail{
 				Type:        evergreen.CommandTypeSystem,
 				Status:      evergreen.TaskFailed,
 				Description: evergreen.TaskDescriptionStranded,
 			}
-			j.AddError(model.TryResetTask(t.Id, evergreen.User, j.ID(), details))
+			catcher.Wrapf(model.TryResetTask(t.Id, evergreen.User, j.ID(), details), "resetting task '%s'", t.Id)
 		}
 	}
 	if len(tasksToDeactivate) > 0 {
 		err = task.DeactivateTasks(tasksToDeactivate, true, j.ID())
-		j.AddError(err)
+		catcher.Wrapf(err, "deactivating tasks exceeding the unschedulable threshold")
 	}
 
-	grip.InfoWhen(!j.HasErrors(),
-		message.Fields{
-			"job":       j.ID(),
-			"op":        j.Type().Name,
-			"num_hosts": len(hosts),
-			"num_tasks": len(taskIDs),
-			"tasks":     taskIDs,
-			"hosts":     hostIDs,
-		})
+	grip.InfoWhen(len(tasks) > 0, message.Fields{
```

Comment:
Can we split this into two grip statements? One that tells us which tasks were in `tasksToDeactivate` and which ones were not?

Reply:
Done.